s2a/src/test/java/io/grpc/s2a/internal/handshaker/FakeS2AServerTest.java

```diff
@@ -118,18 +118,37 @@ public void onCompleted() {
       executor.awaitTermination(1, SECONDS);
     }
 
+    String leafCertString = "";
+    String cert2String = "";
+    String cert1String = "";
+    InputStream leafCert = null;
+    InputStream cert2 = null;
+    InputStream cert1 = null;
+    try {
+      leafCert =
+          FakeS2AServerTest.class.getClassLoader().getResourceAsStream("leaf_cert_ec.pem");
+      cert2 = 
+          FakeS2AServerTest.class.getClassLoader().getResourceAsStream("int_cert2_ec.pem");
+      cert1 = 
+          FakeS2AServerTest.class.getClassLoader().getResourceAsStream("int_cert1_ec.pem");
+      leafCertString = FakeWriter.convertInputStreamToString(leafCert);
+      cert2String = FakeWriter.convertInputStreamToString(cert2);
+      cert1String = FakeWriter.convertInputStreamToString(cert1);
+    } finally {
+      leafCert.close();
```

Comment:
You should do null checks so that if there was a problem with one it doesn't block cleanup of the ones after it.

Reply:
Try-with-resources should be preferred to manual try-finally.

Reply:
Done in https://github.com/grpc/grpc-java/pull/11702/commits/922899d87420d64648441ac2086fabde71e5bf86